presto-hive/src/test/java/com/facebook/presto/hive/TestHiveBucketing.java

```diff
@@ -65,4 +98,231 @@ public void testHashingString()
     {
         return immutableEntry(inspector, value);
     }
+
+    @Test
+    public void testHashingCompare()
```

Comment:
For all of these, I'd remove the bucket count arg, and try it with a bunch of different bucket counts.